master/internal/rm/kubernetesrm/request_workers.go

```diff
@@ -69,6 +81,39 @@ func (r *requestProcessingWorker) receiveCreateKubernetesResources(
 	}
 	r.syslog.Infof("created configMap %s", configMap.Name)
 
+	var gatewayListeners []gatewayTyped.Listener
+	for _, proxyResource := range msg.gatewayProxyResources {
+		r.syslog.Debugf("launching service with spec %v", *proxyResource.serviceSpec)
+		if _, err := r.serviceInterfaces[msg.podSpec.Namespace].Create(
+			context.TODO(), proxyResource.serviceSpec, metaV1.CreateOptions{},
```

Comment:
where would make sense for these contexts to be sourced from in the future?

Reply:
One option is to make request worker have a context that comes from the request queue so we could cancel the request worker and all its requests would be canceled too 

Another option, which I have mentioned to @stoksc, is that it would be cool if ```det e create``` blocks on us submitting resources to k8s, so if there is an issue with your experiment from a k8s perspective you get an immediate error. hpsearch makes this a little challenging but making the system more request scoped could be worth considering.